modules/reactor/src/meshgenerators/PatternedCartesianMeshGenerator.C

```diff
@@ -0,0 +1,753 @@
+                          control_drum_azimuthals[i],
+                          i + 1)); // control drum index to help sort control_drum_id
+    }
+    std::sort(control_drum_tmp.begin(), control_drum_tmp.end());
+    for (unsigned int i = 0; i < control_drum_tmp.size(); ++i)
+    {
+      _control_drum_positions.push_back(std::get<1>(control_drum_tmp[i]));
+      _control_drum_angles.push_back(std::get<0>(control_drum_tmp[i]));
+      _control_drums_azimuthal_meta.push_back(std::get<2>(control_drum_tmp[i]));
+      control_drum_id_sorted.push_back(std::get<3>(control_drum_tmp[i]));
+    }
+
+    if (_assign_control_drum_id)
+    {
+      for (const auto & elem : out_mesh->element_ptr_range())
+      {
+        dof_id_type unsorted_control_drum_id = elem->get_extra_integer(id);
+        if (unsorted_control_drum_id != 0)
+        {
+          auto sorted_iter = std::find(control_drum_id_sorted.begin(),
+                                       control_drum_id_sorted.end(),
+                                       unsorted_control_drum_id);
+          elem->set_extra_integer(id,
+                                  std::distance(control_drum_id_sorted.begin(), sorted_iter) + 1);
+        }
+      }
+    }
+
+    if (_generate_control_drum_positions_file)
+    {
+      std::ofstream pos_file(_position_file_name);
+      for (unsigned int i = 0; i < _control_drum_positions.size(); ++i)
+        pos_file << _control_drum_positions[i](0) << " " << _control_drum_positions[i](1)
+                 << " 0.0\n";
+      pos_file.close();
+    }
+  }
+
+  // Assign customized peripheral block ids and names
+  if (!_peripheral_block_ids.empty())
+    for (const auto & elem : out_mesh->active_element_ptr_range())
+      for (subdomain_id_type i = PERIPHERAL_ID_SHIFT; i <= PERIPHERAL_ID_SHIFT + _duct_sizes.size();
+           i++)
+        if (elem->subdomain_id() == i)
+        {
+          elem->subdomain_id() = _peripheral_block_ids[i - PERIPHERAL_ID_SHIFT];
+          break;
+        }
+  if (!_peripheral_block_names.empty())
+  {
+    for (unsigned i = 0; i < _peripheral_block_names.size(); i++)
+      out_mesh->subdomain_name(_peripheral_block_ids.empty() ? (PERIPHERAL_ID_SHIFT + i)
+                                                             : _peripheral_block_ids[i]) =
+          _peripheral_block_names[i];
+  }
+  // Assign customized outer surface boundary id and name
+  if (_external_boundary_id > 0)
+    MooseMesh::changeBoundaryId(*out_mesh, OUTER_SIDESET_ID, _external_boundary_id, false);
+  if (!_external_boundary_name.empty())
+  {
+    out_mesh->get_boundary_info().sideset_name(
+        _external_boundary_id > 0 ? _external_boundary_id : (boundary_id_type)OUTER_SIDESET_ID) =
+        _external_boundary_name;
+    out_mesh->get_boundary_info().nodeset_name(
+        _external_boundary_id > 0 ? _external_boundary_id : (boundary_id_type)OUTER_SIDESET_ID) =
+        _external_boundary_name;
+  }
+  // assign sideset and nodeset maps
```

Comment:
why is this done? add comment pls

Reply:
Merge the boundary name maps of the input meshes.